tiledb/common/heap_memory.h

```diff
@@ -197,7 +197,7 @@ class tiledb_shared_ptr {
   }
 
   explicit operator bool() const noexcept {
-    return sp_;
+    return static_cast<bool>(sp_);
```

Comment:
Could we do `sp_ ? true : false` instead? I want to make it explicit that we're invoking the sp_ `operator bool()` instead of casting it to a bool.

Reply:
✔️